src/transformers/modeling_tf_utils.py

```diff
@@ -488,15 +488,21 @@ def generate(
 
         Parameters:
 
-            input_ids: (`optional`) `tf.Tensor` of `dtype=tf.int32` of shape `(batch_size, sequence_length)`
+            input_ids: (`optional`) `tf.Tensor` of `dtype=tf.int32 of shape `(batch_size, sequence_length)`
+                Language model: alias for prefix_ids, the sequence used as a prompt for the generation.
+                Seq2seq model: the sequence input to the encoder model.
+
+            prefix_ids: (`optional`) `tf.Tensor` of `dtype=tf.int32 of shape `(batch_size, sequence_length)`
```

Comment:
I think `prefix_ids` should never overwrite `input_ids`. For me it would be ok to add an optional `decoder_input_ids` variable that can be used only  for encoder-decoder models, when the user wants to encode the `input_ids` and then generate from something specific the `decoder_input_ids`. Initially, I didn't see many use cases for this, but maybe there are?? 
Also it might be good to keep the same wording we used for the models:
If  having only "one" model (GPT2), never name anything with `decoder_` . If having an encoder-decoder model, the normal variables "input_ids, attention_mask" are used for the encoder and the decoder uses the same naming and prepends a "decoder_" to the variable names.